### ISSUE TITLE
CompactionStrategy plugin interface not compatible with 2.0.0

### ISSUE THREAD

**Issue by keith-turner** [apache/accumulo#1617]
While working on #1605 I noticed that the pluggable CompactionStrategies in 2.1.0-SNAPSHOT are not compatible with 2.0.0 or 1.9.x.  The CompactionStrategy references a MaJorCompactionRequest type.  This type changed from using FileRef to StoredTabletFile.  This change would break any compaction strategies written by users.  The link below shows the change.  Not sure if we should try to maintain compatibility with this, but did not want to forget about it.

https://github.com/apache/accumulo/commit/94844b3ef88bc9e02b5cb5c8eeff81411ac8221d#diff-ded1031582e4bc5e1aee9c4a3f1c5682

**Comment by milleruntime** [comment]
Good catch.  I can take a look at this.

**Comment by keith-turner** [comment]
@milleruntime it just occurred to me that the fileref type may have also changed.  These plugin interfaces outside of SPI that use internal types are so hard to keep stable because of the transitive need for stability.

**Comment by milleruntime** [comment]
Correct, FileRef was also changed with the introduction of Ample.  At this point I don't know how we can revert FileRef and the CompactionStrategies all the way back to be compatible with 2.0 without significant work.  Especially with all the internal changes that built on top of Ample and your recent changes to compactions.

**Comment by keith-turner** [comment]
We could just mention it in the release notes.

**Comment by milleruntime** [comment]
> We could just mention it in the release notes.

Yeah I like that.  I marked this ticket as a blocker for 2.1 so we remember to add it to the release notes.
